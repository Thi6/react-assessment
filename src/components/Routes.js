import React from 'react';

import { BrowserRouter as Router, Route} from "react-router-dom";

import NavBar from './NavBar';
import Register from './Register';
import Login from './Login';
import UserList from './UserList';



export default function Routes(props) {
    
    return (
        <div>
             
            <Router>
                <div>

                    <NavBar />

                    
                    <Route path="/Register" render={() => <Register getAllUsers={props.getAllUsers} />} />
                    <Route path="/Login" render={() => <Login data={props.data} />} />
                    <Route path="/UserList" render={() => <UserList getAllUsers={props.getAllUsers} data={props.data} />} />


                </div>

      
            </Router>
        </div>
    )

}