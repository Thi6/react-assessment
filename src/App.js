import React, {Component} from 'react';
import axios from 'axios';

import './App.css';
import Routes from './components/Routes';

class App extends Component {
  constructor() {
    super();

    this.state={
      data: []
    }
  }

    componentDidMount() {
        this.getAllUsers();
    }

    getAllUsers = () => {
        axios
        .get('http://localhost:5000/user/all')
        .then(response => {
            
            this.setState({
                data: response.data
            })
            console.log(response);
        })
    }

  render() {
    return (
      <div className="App">
        <Routes getAllUsers={this.state.getAllUsers} data={this.state.data}/>
      </div>
    );
  }

}

export default App;
