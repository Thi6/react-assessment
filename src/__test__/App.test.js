import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import Routes from '../components/Routes';
import App from '../App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<App />)
});

it('renders Routes component', () => {
  const wrapper = shallow(<App />);
  const content = <Routes />
  expect(wrapper.contains(content)).toEqual(true);
});

it("snapshot App", () => {
 // const wrapper = shallow(<App />);
 
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
