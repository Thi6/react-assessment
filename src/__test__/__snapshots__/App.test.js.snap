// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot App 1`] = `
<div
  className="App"
>
  <div>
    <div>
      <div>
        <button>
          <a
            href="/Register"
            onClick={[Function]}
          >
            Register
          </a>
        </button>
        <button>
          <a
            href="/Login"
            onClick={[Function]}
          >
            Login
          </a>
        </button>
        <button>
          <a
            href="/UserList"
            onClick={[Function]}
          >
            Show All Users
          </a>
        </button>
        <br />
      </div>
    </div>
  </div>
</div>
`;
